src/sagemaker/tensorflow/deploying_python.rst

```diff
@@ -0,0 +1,190 @@
+how to make predictions against the Endpoint, how to use different content-types in your requests, and how to extend the Web server
+functionality.
+
+Deploying directly from model artifacts
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you already have existing model artifacts, you can skip training and deploy them directly to an endpoint:
+
+.. code:: python
+
+  from sagemaker.tensorflow import TensorFlowModel
+
+  tf_model = TensorFlowModel(model_data='s3://mybucket/model.tar.gz',
+                             role='MySageMakerRole',
+                             entry_point='entry.py',
+                             name='model_name')
+
+  predictor = tf_model.deploy(initial_instance_count=1, instance_type='ml.c4.xlarge')
+
+You can also optionally specify a pip `requirements file <https://pip.pypa.io/en/stable/reference/pip_install/#requirements-file-format>`_ if you need to install additional packages into the deployed
+runtime environment by including it in your source_dir and specifying it in the ``'SAGEMAKER_REQUIREMENTS'`` env variable:
+
+.. code:: python
+
+  from sagemaker.tensorflow import TensorFlowModel
+
+  tf_model = TensorFlowModel(model_data='s3://mybucket/model.tar.gz',
+                             role='MySageMakerRole',
+                             entry_point='entry.py',
+                             source_dir='my_src', # directory which contains entry_point script and requirements file
+                             name='model_name',
+                             env={'SAGEMAKER_REQUIREMENTS': 'requirements.txt'}) # path relative to source_dir
+
+  predictor = tf_model.deploy(initial_instance_count=1, instance_type='ml.c4.xlarge')
+
+
+Making predictions against a SageMaker Endpoint
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following code adds a prediction request to the previous code example:
+
+.. code:: python
+
+  estimator = TensorFlow(entry_point='tf-train.py', ..., train_instance_count=1,
+                         train_instance_type='ml.c4.xlarge', framework_version='1.10.0')
+
+  estimator.fit(inputs)
+
+  predictor = estimator.deploy(initial_instance_count=1, instance_type='ml.c4.xlarge')
+
+  result = predictor.predict([6.4, 3.2, 4.5, 1.5])
+
+The ``predictor.predict`` method call takes one parameter, the input ``data`` for which you want the ``SageMaker Endpoint``
+to provide inference. ``predict`` will serialize the input data, and send it in as request to the ``SageMaker Endpoint`` by
+an ``InvokeEndpoint`` SageMaker operation. ``InvokeEndpoint`` operation requests can be made by ``predictor.predict``, by
+boto3 ``SageMaker.runtime`` client or by AWS CLI.
+
+The ``SageMaker Endpoint`` web server will process the request, make an inference using the deployed model, and return a response.
+The ``result`` returned by ``predict`` is
+a Python dictionary with the model prediction. In the code example above, the prediction ``result`` looks like this:
```

Comment:
don't need "SageMaker Endpoint" to be monospace

Reply:
ok